FlowCrypt/Controllers/Compose/Extensions/ComposeViewController+MessageSend.swift

```diff
@@ -41,11 +40,44 @@ extension ComposeViewController {
         handleSuccessfullySentMessage()
     }
 
+    internal func requestMissingPassPhraseWithModal(for signingKey: Keypair) {
+        Task {
+            let alert = AlertsFactory.makePassPhraseAlert(
+                onCancel: {
+                    self.handle(error: ComposeMessageError.passPhraseRequired)
+                },
+                onCompletion: { [weak self] passPhrase in
+                    guard let self = self else {
+                        return
+                    }
+                    Task<Void, Never> {
+                        do {
+                            let matched = try await self.composeMessageService.handlePassPhraseEntry(passPhrase, for: signingKey)
+                            if matched {
+                                return continuation.resume(returning: passPhrase)
```

Comment:
@tomholub 
With current refactor, once user enters correct pass phrase, we have to pass entered key to `validateAndProduceSendableMsg` method or somewhere like that.
And then need to check if this is for draft message or send message. (and things like that)

IMHO this makes logic more difficult and it's better to leave as it is now.
Let me know your thought.

Reply:
You don't have to. Instead, you can store the pass phrase in memory (InMemoryPassPhraseService I think) and then retry the whole operation (sending message) without any special arguments. As if user tapped on send again.

The next time it runs, it will pick up the pass phrase from memory. Should be ~ 2 lines of code.

Reply:
That sounds good. Will follow your suggestion 👍